python/simulation.py

```diff
@@ -96,6 +97,8 @@ def get_num_args(func):
     return (
         2
         if isinstance(func, Harminv) or isinstance(func, PadeDFT)
+        else func.__code__.co_argcount - 1  # first argument is "self"
+        if inspect.ismethod(func)
```

Comment:
For improved readability, I would recommend you replace this ternary operator which spans multiple lines with a full `if` statement. 

See: https://google.github.io/styleguide/pyguide.html#2111-definition.